app/controllers/katello/api/v2/sync_plans_controller.rb

```diff
@@ -87,18 +87,12 @@ def destroy
     param_group :search, Api::V2::ApiController
     param :name, String, :desc => N_("product name to filter by")
     def available_products
-      enabled_product_ids = Product.where(:organization_id => @organization).readable.select { |p| p.enabled? }.collect(&:id)
+      product_ids = @sync_plan.product_ids
+      products = Product.enabled.readable
+      products = products.where("#{Product.table_name}.id NOT IN (?)", product_ids) unless product_ids.empty?
 
-      filters = [:terms => {:id => enabled_product_ids - @sync_plan.product_ids}]
-      filters << {:term => {:name => params[:name]}} if params[:name]
-
-      options = {
-        :filters       => filters,
-        :load_records? => true
-      }
-
-      products = item_search(Product, params, options)
-      respond_for_index(:collection => products)
+      options = {:includes => [:sync_plan, :provider], :resource_class => Product}
+      respond_for_index(:collection => scoped_search(products, :name, :desc, options), :template => '../products/index')
```

Comment:
I could to that, but I would have to either:

a) introduce checking for 'available_products' in the URL or action and do some special logic
b) Introduce the available_for concept (https://github.com/Katello/katello/blob/b5f0c408706055e82c04bdd79eef096aee8e84d0/app/controllers/katello/api/v2/repositories_controller.rb#L47) and remove this as an API endpoint through a deprecation cycle